### PR TITLE
Adding support for logo cobranding

### PR DESCRIPTION
**Pull request by JakeBat**
# Problem
We need to add support for logo cobranding similar to the native SDKs

# Solution
Adding an option to enterprise features for logo cobranding behavior similar to that of the native SDKs 

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [x] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: src/components/Router/StepsRouter.tsx [hunk 1]
```diff
@@ -59,13 +60,22 @@ class StepsRouter extends Component<StepsRouterProps> {
       ? cobrand
       : globalUserOptions.enterpriseFeatures?.cobrand && cobrand
 
+    const logoCobrandLogic = mobileFlow
+      ? logoCobrand
+      : globalUserOptions.enterpriseFeatures?.logoCobrand && logoCobrand
+
+    const logoCobrandStyle = logoCobrandLogic
+      ? { backgroundImage: 'url(' + logoCobrandLogic.src + ')' }
```
Comment (kopijunkie) [inline src/components/Router/StepsRouter.tsx:68]:
```suggestion
      ? { backgroundImage: `url(${logoCobrandLogic.src})` }
```

File: src/components/Router/StepsRouter.tsx [hunk 2]
```diff
@@ -59,13 +60,22 @@ class StepsRouter extends Component<StepsRouterProps> {
...
+      : null
+
     return (
       //TODO: Wrap CurrentComponent in themeWrap HOC
       <div
         className={classNames(theme.step, {
           [theme.fullScreenStep]: isFullScreen,
           [theme.noLogo]: hideLogoLogic,
           [theme.cobrandLogo]: cobrandLogic,
+          [theme.logoCobrandLogo]: logoCobrand,
```
Comment (kopijunkie) [inline src/components/Router/StepsRouter.tsx:78]: I think this needs a better/clearer name to distinguish between the "text only" cobrand logo vs this "full image replacement" cobrand logo. If needed, it's OK to rename `cobrandLogo` also.

File: src/types/enterprise.ts [hunk 3]
```diff
@@ -2,7 +2,12 @@ export type EnterpriseCobranding = {
   text: string
 }
 
+export type EnterpriseLogoCobranding = {
+  src: string
+}
+
```
Comment (kopijunkie) [inline src/types/enterprise.ts:8]: Just wondering, why does this logo cobranding need to be a separate config to the existing cobranding one?
Was there issues if it was structured this way, then let the display priority be based on if `src` is defined else use `text`?
```
export type EnterpriseCobranding = {
  text: string
  src: string
}
```

File: src/demo/demoUtils.ts [hunk 4]
```diff
@@ -6,6 +6,8 @@ import {
   StepTypes,
 } from '~types/steps'
 import { ServerRegions, SdkOptions } from '~types/sdk'
+// @ts-ignore
+import testCobrandLogo from './assets/onfido-logo.svg'
```
Comment (kopijunkie) [inline src/demo/demoUtils.ts:10]: Minor: IMO the test cobrand logo image should be something other than the logo that's also used for the existing text only cobranding feature. But appreciate that this also means sourcing a suitable design asset.

File: src/components/Router/StepsRouter.tsx (hunk 1 above)
Comment (kopijunkie) [reply src/components/Router/StepsRouter.tsx:68]: Is there any restrictions/guidelines for the type of image, dimensions that are allowed for the cobrand logo allowed/recommended? If so it should be documented in the README.

**Review by kopijunkie — CHANGES_REQUESTED** [review]

File: src/components/Router/StepsRouter.tsx (hunk 1 above)
Comment (JakeBat) [reply src/components/Router/StepsRouter.tsx:68]: Thanks for the format fix, I've gone ahead and updated it.

As for the image restrictions and documentation, my impression was all enterprise feature related docs should be kept out of the public README, we instead send a separate internal doc upon approval with instructions on how to enable the feature in the code after we've enabled it for their account. I've gone ahead and updated that doc to include the requirements for the image.

File: src/components/Router/StepsRouter.tsx (hunk 2 above)
Comment (JakeBat) [reply src/components/Router/StepsRouter.tsx:78]: Updated cobrandLogo to be textCobrandLogo, let me know if that still isn't clear

File: src/types/enterprise.ts (hunk 3 above)
Comment (JakeBat) [reply src/types/enterprise.ts:8]: I made it a separate object mostly for the validation of the feature in the jwt. `cobrand` and `logoCobrand` are two entirely separate features in that aspect, with `cobrand` meaning text cobranding and `logoCobrand` meaning image cobranding. I would've updated the `EnterpriseCobranding` object to be `EnterpriseTextCobranding` but didn't want to risk any breaking changes for integrators.

File: src/demo/demoUtils.ts (hunk 4 above)
Comment (JakeBat) [reply src/demo/demoUtils.ts:10]: Yeah, I can definitely see that, but I feel making it point to a separate file, even if it's the same image, should be enough coming from a testing perspective. My main concern with using a random image from somewhere is I don't want to create any kind of liability concerns from using an image without permission. If you can think of a good image we can use in it's place we have approval to use, please let me know and I'll switch them out.

File: src/components/Router/StepsRouter.tsx (hunk 1 above)
Comment (kopijunkie) [reply src/components/Router/StepsRouter.tsx]: Ah, ok. That's fine then. Probably just something I missed or perhaps a separate discussion to be had about letting everyone know about the internal doc for the Enterprise Features.

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,7 @@ This project adheres to the Node [default version scheme](https://docs.npmjs.com
 
 - Internal: Added ScreenLayout component. This is currently used in the Welcome and Complete screens.
 - Internal: Added user consent screen
+- Internal: Added logo cobranding enterprise feature
```
Comment (kopijunkie) [inline CHANGELOG.md:15]: Not sure if there's an agreed convention about this, but for a different enterprise feature PR the wording was such that there is a note that the feature was a premium enterprise feature to assumedly make it a bit clearer that not everyone will have access to it.

**Review by kopijunkie — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (JakeBat) [reply CHANGELOG.md]: Nice catch, rewrote the entry in the style hideLogo was

File: src/components/Theme/index.tsx [hunk 6]
```diff
@@ -39,10 +47,17 @@ const themeWrapped = (
...
         </div>
-        {cobrand && !hideOnfidoLogo ? (
-          <div className={classNames({ [theme.cobrandFooter]: cobrand })}>
+        {(cobrand || logoCobrand) && !hideOnfidoLogo ? (
+          <div
+            className={classNames({
+              [theme.cobrandFooter]: cobrand || logoCobrand,
+            })}
+          >
+            {logoCobrand ? <div className={theme.logoCobrandLogo} /> : null}
```
[the remark this answers is not in the document]
Comment (JakeBat) [reply src/components/Theme/index.tsx]: Definitely agree that sounds better, updated.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1354-pr-onfido-sdk-ui-onfido.surge.sh

File: CHANGELOG.md [hunk 7]
```diff
@@ -12,6 +12,7 @@ This project adheres to the Node [default version scheme](https://docs.npmjs.com
 
 - Internal: Added ScreenLayout component. This is currently used in the Welcome and Complete screens.
 - Internal: Added user consent screen
+- Public: Added new enterprise feature `logoCobrand`. When purchased and enabled allows integrator to provide their own logo image to be displayed alongside the onfido logo
```
Comment (kopijunkie) [inline CHANGELOG.md:15]:
```suggestion
- Public: Added new enterprise feature `logoCobrand`. When purchased and enabled allows integrator to provide their own logo image to be displayed alongside the Onfido logo.
```

File: src/components/Router/CrossDeviceMobileRouter.tsx [hunk 8]
```diff
@@ -204,16 +204,29 @@ export default class CrossDeviceMobileRouter extends Component<
...
       ) {
         this.props.actions.hideOnfidoLogo(true)
-      } else {
+      } else if (
+        enterpriseFeatures.cobrand &&
+        validEnterpriseFeatures?.cobrand
+      ) {
         this.props.actions.hideOnfidoLogo(false)
 
         if (enterpriseFeatures.cobrand && validEnterpriseFeatures?.cobrand) {
```
Comment (stefaniacardenas) [inline src/components/Router/CrossDeviceMobileRouter.tsx:213]: You don't need this if statement here anymore

File: src/components/utils/jwt.ts [hunk 9]
```diff
@@ -31,6 +31,8 @@ export const getEnterpriseFeaturesFromJWT = (
 ): EnterpriseFeatures => {
   try {
     const jwt = parseJwt(token)
+    // @ts-ignore
+    jwt.enterprise_features.logoCobrand = true
```
Comment (stefaniacardenas) [inline src/components/utils/jwt.ts:35]: You can just return `{...jwt.enterprise_features, logoCobrand: true}` in the line below and remove this line

Comment (stefaniacardenas) [reply src/components/utils/jwt.ts:35]: So we want the `logoCobrand` to be true for all `enterprise_features`?

File: src/components/Router/CrossDeviceMobileRouter.tsx (hunk 8 above)
Comment (JakeBat) [reply src/components/Router/CrossDeviceMobileRouter.tsx:213]: Thanks, updated.

File: src/components/utils/jwt.ts (hunk 9 above)
Comment (JakeBat) [reply src/components/utils/jwt.ts:35]: Lol my apologies, I just did this for testing, thanks for catching! Updated.

**Review by stefaniacardenas — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1354-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1354-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**